pkg/authorization/authorizer/scope/converter.go

```diff
@@ -244,6 +300,37 @@ func (e clusterRoleEvaluator) ResolveRules(scope, namespace string, clusterPolic
 	return rules, nil
 }
 
+func (e clusterRoleEvaluator) ResolveGettableNamespaces(scope string, clusterPolicyGetter client.ClusterPolicyLister) ([]string, error) {
+	_, scopeNamespace, _, err := e.parseScope(scope)
+	if err != nil {
+		return nil, err
+	}
+	rules, err := e.resolveRules(scope, clusterPolicyGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	attributes := authorizer.DefaultAuthorizationAttributes{
+		APIGroup: kapi.GroupName,
+		Verb:     "get",
+		Resource: "namespaces",
```

Comment:
> normally, there's a ResourceName in the attributes we pass to RuleMatches()... trying to reason about what it means when there isn't one

If a rule has no resource names, it can match.  If the rule has a name, someone would have had to create a rule with `name == ""`.  I can update validation to prevent that in new roles if you like, but its a pretty insane thing to do.

Most rules won't have names on namespace rules.  Those that do will be honored.  Those that don't behave properly.